Add 0.71 blog post (#3513)
facebook/react-native-website: website/blog/2023-01-12-version-071.md

```js
import ReactNativeFeatureFlags from 'react-native/Libraries/ReactNative/ReactNativeFeatureFlags';

// enable the JS-side of the w3c PointerEvent implementation
ReactNativeFeatureFlags.shouldEmitW3CPointerEvents = () => true;

// enable hover events in Pressibility to be backed by the PointerEvent implementation.
// shouldEmitW3CPointerEvents should also be true
ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover =
  () => true;
```

:::note

You’ll also need to enable React feature flags on your [Android](https://reactnative.dev/blog/2022/12/13/pointer-events-in-react-native#android-specific) and [iOS](https://reactnative.dev/blog/2022/12/13/pointer-events-in-react-native#ios-specific) native setup.

:::

Check out our dedicated [PointerEvents post](https://reactnative.dev/blog/2022/12/13/pointer-events-in-react-native) to learn more.

## Restoring PropTypes

React Native’s prop types, such as `ViewPropTypes` and `Text.propTypes`, were deprecated in 0.66 and accessing them would output deprecation warnings. When they were removed in 0.68, many developers began experiencing errors when upgrading to the latest version of React Native.

> After some investigation, we realized that a couple issues prevented the community from taking action on the deprecation warnings. First, the deprecation warning was not always actionable which caused people to ignore them ([issue one](https://facebook/react-native#34650), [issue two](https://github.com/react-native-community/cli#1699)). Second, the deprecation warnings were being [incorrectly filtered](https://facebook/react-native@fa2842d) by `LogBox.ignoreLogs`. Both of these have now been fixed, but we want to give people more time to upgrade the deprecated call sites.

Reviewer: Couple of the links here are broken

> After some investigation, we realized that a couple issues prevented the community from taking action on the deprecation warnings. First, the deprecation warning was not always actionable which caused people to ignore them ([issue one](https://facebook/react-native#34650), [issue two](https://github.com/react-native-community/cli#1699)). Second, the deprecation warnings were being [incorrectly filtered](https://facebook/react-native@fa2842d) by `LogBox.ignoreLogs`. Both of these have now been fixed, but we want to give people more time to upgrade the deprecated call sites.

Reviewer: ```suggestion
After some investigation, we realized that a couple issues prevented the community from taking action on the deprecation warnings. First, the deprecation warning was not always actionable which caused people to ignore them ([issue one](https://github.com/facebook/react-native/pull/34650), [issue two](https://github.com/react-native-community/cli/pull/1699)). Second, the deprecation warnings were being [incorrectly filtered](https://github.com/facebook/react-native/commit/fa2842d) by `LogBox.ignoreLogs`. Both of these have now been fixed, but we want to give people more time to upgrade the deprecated call sites.
```

---
title: 'React Native 0.71: TypeScript by Default, Flexbox Gap, and more...'

> authors: [lunaleaps, mattcarrollcode, cortinico, kelset]

Reviewer 2: historically, we've put the entire release crew as authors of the release blogpost per each release:
- https://reactnative.dev/blog/2022/09/05/version-070
- https://reactnative.dev/blog/2022/06/21/version-069
- https://reactnative.dev/blog/2022/03/30/version-068
- https://reactnative.dev/blog/2022/01/19/version-067

But I understand that a lot of work was done by other folks too. A more realistic authors tag would likely be:
```suggestion
authors: [lunaleaps, mattcarrollcode, cortinico, kelset, dmytrorykun, cipolleschi, titozzz, rickhanlonii]
```

but `rickhanlonii` needs to be added to https://github.com/facebook/react-native-website/blob/main/website/blog/authors.yml

Reviewer: Comment from Rick that was on internal:

> I wasn't sure who exactly to put in as the authors, so currently it's [Luna](https://fb.workplace.com/profile.php?id=100029844272183), [Matt](https://fb.workplace.com/profile.php?id=100085011314613), Lorenzo, and [Nicola](https://fb.workplace.com/profile.php?id=100068089718865). In the future I'd like to be able to include more/all authors, or author posts as a team, but with the current design taking up so much vertical space, I think we should limit it to 4 or less people so the content isn't pushed too far down.

Author: Added a thanks to the release crew in the Acknowledgements. I'm happy to just be a ghostwriter for this release to cut down on the space the author section takes up.

tags: [announcement, release]
date: 2023-01-12
---

Today we’re releasing React Native version 0.71. This is a feature-packed release which includes TypeScript by default, Flexbox gap support, new web-inspired props, developer experience improvements, updates to the New React Native Architecture, and more!

> 

Reviewer 3: It probably makes sense to truncate here and link to the different titles
```suggestion

<!--truncate-->

```

Reviewer 2: here's a reference of how it should be https://github.com/facebook/react-native-website/blob/main/website/blog/2023-01-03-typescript-first.md?plain=1#L16

Reviewer 2: titles that should be listed pre-truncate:
* TypeScript by default
* Simplifying layouts with Flexbox gap
* Web-inspired props for accessibility, styles, and events
* Restoring PropTypes
* Developer Experience Improvements
* Breaking changes

> Today we’re releasing React Native version 0.71. This is a feature-packed release which includes TypeScript by default, Flexbox gap support, new web-inspired props, developer experience improvements, and improvement to the New Architecture experimental experience.

Reviewer 4: > New Architecture experimental experience

This doesn't really roll off the tongue. I know there was a bunch of debate about "experimental", so if you want to include all those words, how about "experimental New Architecture experience"

Author: Updated, what do you think?

## TypeScript by default

In this release, we’re investing in the TypeScript experience of React Native.

Starting with 0.71, when you create a new React Native app via the React Native CLI you'll get a TypeScript app by default. The new project is already set up with a `tsconfig.json` so out of the box your IDE will help you write typed code right away.

> We’re also offering built-in TypeScript declarations directly from the `react-native` package. This means you won’t need `@types/react-native` any longer.

Reviewer 4: Do you want to throw in "more accurate" or something like that?

> offering built-in, more accurate TypeScript declarations...

Because the key point isn't just that they need one less npm package, right? It's that the quality of the definitions should be better now and going forward.

Finally, our documentation has been updated to feature TypeScript for all examples.

:::note

After upgrading to React Native 0.71, we recommend removing `@types/react-native` from your package.json `devDependencies`.

:::

For more details on this change, including migration steps and how this affects Flow users, check out our previous post [_First-class Support for TypeScript_](https://reactnative.dev/blog/2023/01/03/typescript-first).

## Simplifying layouts with Flexbox gap

With React Native you can flexibly layout components on different screen sizes using flexbox. Browsers have supported the flexbox properties [_gap_](https://developer.mozilla.org/en-US/docs/Web/CSS/gap), [_rowGap_](https://developer.mozilla.org/en-US/docs/Web/CSS/row-gap), and [_columnGap_](https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap), which allow you to specify the amount of space between all items in a flexbox.

> These properties have been long requested in React Native, and 0.71 adds initial support for gaps defined using pixel values. In future version, we will add support for more values, such as percentages.

Reviewer 3: ```suggestion
These properties have been long requested in React Native, and 0.71 adds initial support for gaps defined using pixel values. In future versions, we will add support for more values such as percentages.
```

To see why this is useful, imagine trying to build a responsive layout with variably sized cards, 10px apart from each other, hugging the edges of the parent container. Trying to accomplish this layout with child margins can be tricky.

The following shows a layout where we start by giving each child `margin: 10`:

![Two diagrams. On the left it shows a skeleton of an app with three boxes that have margin around them cause the boxes to have margin around all sides. On the right, the same diagram is shown highlighted to demonstrate the margin on all sides.](/blog/assets/FlexboxGapBefore.png)

> Margins are applied uniformly to the edges of all children and don’t collapse under flexbox, giving us spacing at the exterior of the cards, and double the space on the interior compared to what we wanted. We can get around this by applying non-uniform margins, using negative margins on the parent, halving our intended spacing, etc, but it can be made much easier.

Reviewer: ```suggestion
Margins are applied uniformly to the edges of all children and don’t collapse under Flexbox, giving us spacing at the exterior of the cards, and double the space on the interior compared to what we wanted. We can get around this by applying non-uniform margins, using negative margins on the parent, halving our intended spacing, etc, but it can be made much easier.
```

> With flex gap, this layout can be achieved by setting `gap: 10` for a 10px gap between the interior of the cards:

Reviewer: ```suggestion
With flex gap, this layout can be achieved by setting `gap: 10` on the container for a 10px gap between the interior of the cards:
```

Realizing I left some ambiguity here on whether to apply to child or container.

> With flex gap, this layout can be achieved by setting `gap: 10` for a 10px gap between the interior of the cards:

Reviewer 5: ```suggestion
With flex gap, this layout can be achieved by setting `gap: 10` on the container for a 10 pixel gap between the interior of the cards:
```

Phrasing nit including Nick addition.

> ![[Two diagrams. On the left it shows a skeleton of an app with three boxes that have margin only on the inner sides and not the outer sides of the boxes due to the Flexbox gap property. On the right, the same diagram is shown highlighted to demonstrate the margin only on the inner sides.](/blog/assets/FlexboxGapAfter.png)

Reviewer 6: extra [ breaks markdown here

Reviewer 2: ```suggestion
![Two diagrams. On the left it shows a skeleton of an app with three boxes that have margin only on the inner sides and not the outer sides of the boxes due to the Flexbox gap property. On the right, the same diagram is shown highlighted to demonstrate the margin only on the inner sides.](/blog/assets/FlexboxGapAfter.png)
```

> For more information on Flexbox gaps, see [_https://css-tricks.com/minding-the-gap/#aa-flexbox-gaps_](https://css-tricks.com/minding-the-gap/#aa-flexbox-gaps).

Reviewer 2: ```suggestion
For more information on Flexbox gaps, see [this dedicated blogpost from CSS Tricks](https://css-tricks.com/minding-the-gap/#aa-flexbox-gaps).
```

## Web-inspired props for accessibility, styles, and events

:::note

> For more background on aligning React Native to web standards, check out this [_proposal_](https://github.com/necolas/discussions-and-proposals/blob/reduce-fragmentation/proposals/0000-reduce-fragmentation.md) and [_related discussion_](https://github.com/react-native-community/discussions-and-proposals/pull/496).

Reviewer 4: The other sections that have a note have it at the bottom while this one has it at the top. Should it be at the bottom so we don't send people to another link before the read what's here?

:::

This release includes a number of new props inspired by web standards to align React Native’s APIs across many platforms. These new props are purely additive so there are no expected migrations or change of behavior for equivalent accessibility, behavior, or style props.

> For any new prop alias introduced, if there is an existing prop with a different name and both are specified, the new alias prop value will take precedence. For example, this release adds a `src` prop alias for `source` on the Image component to align with the `src` prop on web. If both `src` and `source` are provided, the new `src` prop will be used.

Reviewer 6: it is not very clear to me whether it’s recommended to use the new props or not. as in — are they going to eventually replace the old ones? coexist forever?

Author: It's not clear to me what the recommendation is, or if we plan to remove the old props in a future version. @necolas @lunaleaps do you know thee answer?

Reviewer 7: Small heads up that we are going to publish the blogpost leaving this as unclear for now. It could require longer discussion and we could update the blogpost once we have more clarity on how to proceed.

#### Accessibility

We introduced ARIA props as alias to existing React Native accessibility props.

These props now exist on all core components of React Native:

> - `aria-label`

Reviewer 4: Did these land with dashes in the name? For some reason I thought they landed with camel case like `ariaLabel`

Reviewer 3: Oh good call out, just checked and they were landed with dashes.

Author: Yeah, including the dash matches web https://reactjs.org/docs/accessibility.html#wai-aria

- `aria-labelledby`
- `aria-modal`
- `id`
- `aria-busy`
- `aria-checked`
- `aria-disabled`
- `aria-expanded`
- `aria-selected`
- `aria-valuemax`
- `aria-valuemin`
- `aria-valuenow`
- `aria-valuetext`

We also introduced equivalent web behavior for the following props:

- `aria-hidden`
- `aria-live`
- `role`
- `tabIndex`

See this [_issue_](https://github.com/facebook/react-native/issues/34424) for more details.

#### Component-Specific Behavior

There were also props introduced to align prop names with equivalent DOM prop names for core components.

**Image**

- `alt`
- `tintColor`
- `crossOrigin`
- `height`
- `referrerPolicy`
- `src`
- `srcSet`
- `width`

**TextInput**

- `autoComplete`

> - `enterKeyHint

Reviewer 4: Missing a `

- `inputMode`
- `readOnly`
- `rows`

See this [_issue_](https://github.com/facebook/react-native/issues/34424) for more details.

#### Styles

To align with certain CSS styles, there have been feature extensions for the following styles:

> - [`aspectRatio`](https://reactnative.dev/docs/next/layout-props#aspectratio) now supports string values

Reviewer 3: We link some of the props to the docs added but not all. I think ideally we should link all props mentioned here to their docs.

However, I'm wondering if we should wait until 0.71 docs are cut over, and then add the links. Since `next` will be wrong. So for now, should we remove the links for all these?

Reviewer 2: > However, I'm wondering if we should wait until 0.71 docs are cut over, and then add the links -- since next will be wrong. So for now, should we remove the links for all these?

One workaround would be to just remove the `/next` portion of the link, since once we merge the docs cut PR 0.71 will be current version so the link will all be as such (ex. https://reactnative.dev/docs/height-and-width) and we'll merge first the docs cut, then the blogpost PR. (as in, no need to wait for the 0.71 docs to have rolled out, we can "pre-mock" the urls)

- `fontVariant` now supports space-separated string values
- `fontWeight` now supports number values
- `transform` now supports string values

The following aliases have been added to shadow existing React Native styles:

- [`objectFit`](https://reactnative.dev/docs/next/image#objectfit)
- `pointerEvents`
- `userSelect`
- [`verticalAlign`](https://reactnative.dev/docs/next/text-style-props#verticalalign-android)

See this [_issue_](https://github.com/facebook/react-native/issues/34425) for more details.

#### Events

In this release we also added an opt-in implementation of [_PointerEvents_](https://w3c.github.io/pointerevents)

Once enabled, the following handlers on `View` will support hover:

- `onPointerOver`, `onPointerOut`
- `onPointerEnter`, `onPointerLeave`

These events are also implemented in `Pressability` for new opt-in support for hover.

To enable these features, set the following flags to true:

> ```

Reviewer 5: ```suggestion
```jsx
```

Missing code block language.

import ReactNativeFeatureFlags from "react-native/Libraries/ReactNative/ReactNativeFeatureFlags";

// enable the JS-side of the w3c PointerEvent implementation
ReactNativeFeatureFlags.shouldEmitW3CPointerEvents = () => true;

// enable hover events in Pressibility to be backed by the PointerEvent implementation.
// shouldEmitW3CPointerEvents should also be true

> ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover = () =>
>   true;
> 

Reviewer 5: ```suggestion
ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover = () => true;
```

Remove additional whitespace from code block.

> ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover = () =>
>   true;
> 
> ```

Reviewer 5: ```suggestion
ReactNativeFeatureFlags.shouldPressibilityUseW3CPointerEventsForHover = () => true;
```
```

Remove additional whitespace.

:::

Check out our dedicated [_PointerEvents post_](https://reactnative.dev/blog/2022/12/13/pointer-events-in-react-native) to learn more.

## Restoring PropTypes

React Native’s prop types, such as `ViewPropTypes` and `Text.propTypes`, were deprecated in 0.66 and accessing them would output deprecation warnings. When they were removed in 0.68, many developers began experiencing errors when upgrading to the latest version of React Native.

After some investigation, we realized that a couple issues prevented the community from taking action on the deprecation warnings. First, the deprecation warning was not always actionable which caused people to ignore them ([issue one](https://facebook/react-native#34650), [issue two](https://github.com/react-native-community/cli#1699)). Second, the deprecation warnings were being [incorrectly filtered](https://facebook/react-native@fa2842d) by `LogBox.ignoreLogs`. Both of these have now been fixed, but we want to give people more time to upgrade the deprecated call sites.

> So in this release we are adding back React Native’s prop types so that it is easier for people to upgrade and migrate their code to avoid using them. The `deprecated-react-native-prop-types` package has also been updated for all of the props in 0.71. In the future, we plan to proceed with the deprecation and remove prop types once again. We expect that when we revisit the removal, the community will experience significantly less issues.

Reviewer 4: prop types -> propTypes ?

Reviewer 2: ```suggestion
So in this release we are adding back React Native’s propTypes so that it is easier for people to upgrade and migrate their code to avoid using them. The `deprecated-react-native-prop-types` package has also been updated for all of the props in 0.71. In the future, we plan to proceed with the deprecation and remove prop types once again. We expect that when we revisit the removal, the community will experience significantly less issues.
```

:::caution

> As part of this change, we are also removing the console filtering from `LogBox.ignoreLog`. This means logs that you have previously filtered with `Logbox.ignoreLog` will start apearing again in the console when you upgrade.

Reviewer 2: ```suggestion
As part of this change, we are also removing the console filtering from `LogBox.ignoreLog`. This means logs that you have previously filtered with `Logbox.ignoreLog` will start appearing again in the console when you upgrade.
```

This is expected, so that since warnings and errors such as deprecation warnings can be found and fixed.
:::

## Developer Experience Improvements

### React DevTools

In this release, we've brought two popular React DevTools features on web to React Native.

> Click to inspect is the option in the top left corner of React Dev Tools that allows you to click on an item in the app to inspect it in Dev Tools, similar to the Chrome element inspector.

Reviewer 3: I think grouping "Click to inspect" will make this sentence clearer.
```suggestion
"Click to inspect" is the option in the top left corner of React Dev Tools that allows you to click on an item in the app to inspect it in Dev Tools, similar to the Chrome element inspector.
```

Component highlighting will highlight the element you select in DevTools in the app so you can see which React components line up with which on-screen elements.

Here are both features in action:

![Video of the behavior described above in action. On the left is a React Native app running in an iPhone simulator. On the right is the React DevTools. In both workflows, clicking on an item in the DevTools highlights the corresponding components in the app.](/blog/assets/ElementInspecting.gif)

### Hermes

In React Native 0.70, we made [Hermes the default engine for React Native](https://reactnative.dev/blog/2022/07/08/hermes-as-the-default).

In React Native 0.71, we’re upgrading Hermes with a few notable improvements:

- **Improve source maps**: By loading source maps over the network with Metro we’ve restored the ability to use source maps in recent versions of Chrome Dev Tools outside of Flipper.
- **Improve `JSON.parse` performance**: This version includes a performance optimization that improves the performance of `JSON.parse` up to 30%.

> - **Add support for `.at()`**: Hermes now [_includes support requested_](https://github.com/facebook/hermes/issues/823) for the `.at()` method for `String`, `TypedArray`, and `Array`.

Reviewer 4: This reads as though `at` was already support but just missing these features. Is that correct? I can't tell from the linked post.

If not, could this just be:

Hermes now supports `.at()` on String, TypedArray, and Array.

For a full list of changes see [_the Road to 71 issue_](https://github.com/reactwg/react-native-releases/discussions/41#discussioncomment-4089256).

### New Architecture

> This release brings many improvements to the experiments New Architecture experience based on user feedback and reports we collected so far.

Reviewer 4: Experiments -> experimental

Reviewer 2: ```suggestion
This release brings many improvements to the experimental New Architecture experience based on user feedback and reports we collected so far.
```

- **Reduced build times**: the new distribution model uses Maven Central, which allows us to greatly reduce the build time on Android, resolves many build problems on Windows, and provides a more seamless experience with the New Architecture. [_Read more here_](https://github.com/reactwg/react-native-new-architecture/discussions/105).

> - **Write less C++ code**: you can now enable the New Architecture without having to add any C++ code in your app and the CLI app template has been cleaned of all the C++ code and the CMake files. [_React more here_](https://github.com/reactwg/react-native-new-architecture/discussions/101).

Reviewer 6: typo: React more => Read more

Reviewer 2: ```suggestion
- **Write less C++ code**: you can now enable the New Architecture without having to add any C++ code in your app and the CLI app template has been cleaned of all the C++ code and the CMake files. [_Read more here_](https://github.com/reactwg/react-native-new-architecture/discussions/101).
```

### Other Notable Fixes

- **Better stack frame collapsing**: We've [updated the list of internal frames](https://github.com/react-native-community/cli/pull/1699) for React Native so LogBox will show your code more often rather than internal React Native frames, helping you to debug issues faster.
- **Build time improvements:** We migrated assets to Maven Central for prefabs to improve build times (both iOS and Android) for Hermes in both the current and new architectures.
- **Android template improvements**: The Android template was heavily cleaned and now fully relies on the React Native Gradle Plugin. You can find the configuration instructions directly inside the template or in the [_new dedicated page on the website_](https://reactnative.dev/docs/next/react-native-gradle-plugin).

## Breaking changes

- **Changes to Console Logging:** `LogBox.ignoreLog` no longer filters console logs. This means you will start seeing logs in the console that you have silenced in LogBox. See [this comment](https://github.com/facebook/react-native/pull/34476#issuecomment-1240667794) for more details.

> - **Removed AsyncStorage and MaskedViewIOS**: These components have been deprecated since version (TODO), so it’s time we remove them entirely. For alternatives, please check [_React Native Directory_](https://reactnative.directory/) for community packages that cover those use cases.

Reviewer 4: This todo needs to be filled in. Anyone know the version number to add?

Reviewer 2: 0.59: https://github.com/facebook/react-native/blob/main/CHANGELOG.md#deprecated-8

Reviewer 2: ```suggestion
- **Removed AsyncStorage and MaskedViewIOS**: These components have been deprecated since version [0.59](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#deprecated-8), so it’s time we remove them entirely. For alternatives, please check [_React Native Directory_](https://reactnative.directory/) for community packages that cover those use cases.
```

- **JSCRuntime moved to react-jsc:** react-jsi is now split into react-jsc and react-jsi. If you use JSCRuntime, you will need to add react-jsc as a dependency ([facebook/react-native@6b129d8](https://github.com/facebook/react-native/commit/6b129d81ed8cab301775d2a04971e255df9290de)).

## Acknowledgements

This release is possible thanks to the work of 70+ contributors adding over 1000 commits.

We especially want to thank those who contributed to these major React Native projects:

- **Flexbox Gap Support**: [@intergalacticspacehighway](https://github.com/intergalacticspacehighway) and [@jacobp100](https://github.com/jacobp100).
- **Web-inspired props**: [@gabrieldonadel](https://github.com/gabrieldonadel) [@dakshbhardwaj](https://github.com/dakshbhardwaj) [@dhruvtailor7](https://github.com/dhruvtailor7) [@ankit-tailor](https://github.com/ankit-tailor) [@madhav23bansal](https://github.com/madhav23bansal).

> - **Codegen Improvements**: [_@AntoineDoubovetzky_](http://github.com/AntoineDoubovetzky), [_@MaeIg_](http://github.com/MaeIg), [_@Marcoo09_](http://github.com/Marcoo09), [_@Naturalclar_](http://github.com/Naturalclar), [_@Pranav-yadav_](http://github.com/Pranav-yadav), [_@ZihanChen-MSFT_](http://github.com/ZihanChen-MSFT), [_@dakshbhardwaj_](http://github.com/dakshbhardwaj), [_@dhruvtailor7_](http://github.com/dhruvtailor7), [_@gabrieldonadel_](http://github.com/gabrieldonadel), [_@harshsiri110_](http://github.com/harshsiri110), [_@ken0nek_](http://github.com/ken0nek), [_@kylemacabasco_](http://github.com/kylemacabasco), [_@matiassalles99_](http://github.com/matiassalles99), [_@mdaj06_](http://github.com/mdaj06), [_@mohitcharkha_](http://github.com/mohitcharkha), [_@tarunrajput_](http://github.com/tarunrajput), [_@vinayharwani13_](http://github.com/vinayharwani13), [_@youedd_](http://github.com/youedd).

Reviewer 2: some links here have the _ italic, some don't.

## Try out 0.71.0 now!

For React Native CLI users, see the [_upgrade documentation_](https://reactnative.dev/docs/upgrading) for how to update your existing project, or create a new project with `npx react-native init MyProject`.

React Native version 0.71 will be supported in Expo SDK version 48.

> _Note: 0.71 is now the latest stable version of React Native and 0.68.x versions are now unsupported. For more information see [React Native’s support policy](https://github.com/reactwg/react-native-releases#releases-support-policy)._

Reviewer 5: ```suggestion
:::info
0.71 release is now the latest stable version of React Native and 0.68.x versions are now unsupported. For more information see [React Native’s support policy](https://github.com/reactwg/react-native-releases#releases-support-policy)._
:::
```

We should always use annotation before blockquotes for marking content parts. Blockquote was an alternative when we do not have the admonitions, and in a long term will lose the styling and become the quotes they should be.
